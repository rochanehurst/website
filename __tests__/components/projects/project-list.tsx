import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ProjectList from "../../../components/main/projects/project-list";

const exampleProjects = [
  {
    title: "Test Project 1",
    description: "Test Description 1",
    websiteUrl: "",
    githubUrl: "",
    imageSrc: "/images/projects/routify.gif",
    tags: ["JavaScript", "NextJS", "ReactJS"],
    date: "Jan 2024 - Mar 2024",
  },
  {
    title: "Test Project 2",
    description: "Test Description 2",
    websiteUrl: "",
    githubUrl: "",
    imageSrc: "/images/projects/routify.gif",
    tags: ["JavaScript", "NextJS", "ReactJS"],
    date: "Jan 2024 - Mar 2024",
  },
];

describe('ProjectList', () => {
  it('renders without crashing', () => {
    render(<ProjectList projects={exampleProjects} />);

    for (const project of exampleProjects) {
      expect(screen.getByText(project.title)).toBeInTheDocument();
      expect(screen.getByText(project.description)).toBeInTheDocument();

      // Check for multiple matching date elements
      const dateElements = screen.getAllByText(project.date);
      expect(dateElements.length).toBeGreaterThan(0);
    }
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ProjectList projects={exampleProjects} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
