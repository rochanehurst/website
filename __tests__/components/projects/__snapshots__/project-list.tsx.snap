// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      class="text-2xl font-bold text-foreground/80 w-full text-center sm:text-left px-2 md:px-0 mb-8"
    >
      Projects
    </h1>
    <div
      class="w-full flex flex-row flex-wrap gap-[1rem] px-2 md:px-0"
    >
      <div
        class="rounded-md border border-border w-full overflow-hidden custom-box-shadow sm:!w-[calc(50%-1rem)] md:!w-[calc(33.333%-1rem)]"
      >
        <div
          class="w-full overflow-hidden h-[180px] border-b border-border"
        >
          <img
            alt="content-image"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            fetchpriority="auto"
            height="1080"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fprojects%2Froutify.gif&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Fprojects%2Froutify.gif&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fprojects%2Froutify.gif&w=3840&q=75 2x"
            style="color: transparent;"
            width="1920"
          />
        </div>
        <div
          class="w-full p-4 flex flex-col dark:bg-primary-foreground min-h-0"
        >
          <div>
            <div
              class="space-y-2 line-clamp-6"
            >
              <h1
                class="font-bold text-foreground/90 line-clamp-1"
              >
                Test Project 1
              </h1>
              <span
                class="flex gap-2 items-center"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="Calendar"
                  >
                    <path
                      d="M18.438,4.954H16.5c0-0.346,0-0.691,0-1.036c0-0.124,0-0.248,0-0.372c0-0.262-0.23-0.512-0.5-0.5
            c-0.271,0.012-0.5,0.22-0.5,0.5c0,0.469,0,0.939,0,1.408h-7c0-0.346,0-0.691,0-1.036c0-0.124,0-0.248,0-0.372
            c0-0.262-0.23-0.512-0.5-0.5c-0.271,0.012-0.5,0.22-0.5,0.5c0,0.469,0,0.939,0,1.408H5.562c-1.378,0-2.5,1.122-2.5,2.5v11
            c0,1.379,1.122,2.5,2.5,2.5h12.875c1.379,0,2.5-1.121,2.5-2.5v-11C20.938,6.076,19.816,4.954,18.438,4.954z M5.562,5.954H7.5
            c0,0.073,0,0.147,0,0.22c0,0.124,0,0.248,0,0.372c0,0.262,0.23,0.512,0.5,0.5c0.271-0.012,0.5-0.22,0.5-0.5c0-0.197,0-0.394,0-0.592
            h7c0,0.073,0,0.147,0,0.22c0,0.124,0,0.248,0,0.372c0,0.262,0.23,0.512,0.5,0.5c0.271-0.012,0.5-0.22,0.5-0.5
            c0-0.197,0-0.394,0-0.592h1.937c0.827,0,1.5,0.673,1.5,1.5v1.584H4.062V7.454C4.062,6.627,4.735,5.954,5.562,5.954z M18.438,19.954
            H5.562c-0.827,0-1.5-0.673-1.5-1.5v-8.416h15.875v8.416C19.938,19.281,19.265,19.954,18.438,19.954z"
                    />
                  </g>
                </svg>
                <h2
                  class="text-sm text-foreground/80"
                >
                  Jan 2024 - Mar 2024
                </h2>
              </span>
              <p
                class="text-sm text-foreground/70"
              >
                Test Description 1
              </p>
            </div>
            <div
              class="flex flex-wrap gap-2 my-4"
            >
              <span
                class="p-1 rounded-sm bg-primary-foreground dark:bg-secondary-foreground/20 text-xs"
              >
                JavaScript
              </span>
              <span
                class="p-1 rounded-sm bg-primary-foreground dark:bg-secondary-foreground/20 text-xs"
              >
                NextJS
              </span>
              <span
                class="p-1 rounded-sm bg-primary-foreground dark:bg-secondary-foreground/20 text-xs"
              >
                ReactJS
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="rounded-md border border-border w-full overflow-hidden custom-box-shadow sm:!w-[calc(50%-1rem)] md:!w-[calc(33.333%-1rem)]"
      >
        <div
          class="w-full overflow-hidden h-[180px] border-b border-border"
        >
          <img
            alt="content-image"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            fetchpriority="auto"
            height="1080"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fprojects%2Froutify.gif&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Fprojects%2Froutify.gif&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fprojects%2Froutify.gif&w=3840&q=75 2x"
            style="color: transparent;"
            width="1920"
          />
        </div>
        <div
          class="w-full p-4 flex flex-col dark:bg-primary-foreground min-h-0"
        >
          <div>
            <div
              class="space-y-2 line-clamp-6"
            >
              <h1
                class="font-bold text-foreground/90 line-clamp-1"
              >
                Test Project 2
              </h1>
              <span
                class="flex gap-2 items-center"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="Calendar"
                  >
                    <path
                      d="M18.438,4.954H16.5c0-0.346,0-0.691,0-1.036c0-0.124,0-0.248,0-0.372c0-0.262-0.23-0.512-0.5-0.5
            c-0.271,0.012-0.5,0.22-0.5,0.5c0,0.469,0,0.939,0,1.408h-7c0-0.346,0-0.691,0-1.036c0-0.124,0-0.248,0-0.372
            c0-0.262-0.23-0.512-0.5-0.5c-0.271,0.012-0.5,0.22-0.5,0.5c0,0.469,0,0.939,0,1.408H5.562c-1.378,0-2.5,1.122-2.5,2.5v11
            c0,1.379,1.122,2.5,2.5,2.5h12.875c1.379,0,2.5-1.121,2.5-2.5v-11C20.938,6.076,19.816,4.954,18.438,4.954z M5.562,5.954H7.5
            c0,0.073,0,0.147,0,0.22c0,0.124,0,0.248,0,0.372c0,0.262,0.23,0.512,0.5,0.5c0.271-0.012,0.5-0.22,0.5-0.5c0-0.197,0-0.394,0-0.592
            h7c0,0.073,0,0.147,0,0.22c0,0.124,0,0.248,0,0.372c0,0.262,0.23,0.512,0.5,0.5c0.271-0.012,0.5-0.22,0.5-0.5
            c0-0.197,0-0.394,0-0.592h1.937c0.827,0,1.5,0.673,1.5,1.5v1.584H4.062V7.454C4.062,6.627,4.735,5.954,5.562,5.954z M18.438,19.954
            H5.562c-0.827,0-1.5-0.673-1.5-1.5v-8.416h15.875v8.416C19.938,19.281,19.265,19.954,18.438,19.954z"
                    />
                  </g>
                </svg>
                <h2
                  class="text-sm text-foreground/80"
                >
                  Jan 2024 - Mar 2024
                </h2>
              </span>
              <p
                class="text-sm text-foreground/70"
              >
                Test Description 2
              </p>
            </div>
            <div
              class="flex flex-wrap gap-2 my-4"
            >
              <span
                class="p-1 rounded-sm bg-primary-foreground dark:bg-secondary-foreground/20 text-xs"
              >
                JavaScript
              </span>
              <span
                class="p-1 rounded-sm bg-primary-foreground dark:bg-secondary-foreground/20 text-xs"
              >
                NextJS
              </span>
              <span
                class="p-1 rounded-sm bg-primary-foreground dark:bg-secondary-foreground/20 text-xs"
              >
                ReactJS
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
